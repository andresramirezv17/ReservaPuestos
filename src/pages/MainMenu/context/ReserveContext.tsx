import React, { createContext, useEffect, useState } from 'react';
import moment from 'moment';
import { SearchParameters } from './models/SearchParameters';
import { WorkspacesSearched } from './models/WorkspacesSearched';
import { Floors } from './models/Floors';
import { Sections } from './models/Sections';
import * as ReserveServices from '../services/WorkspaceServices';

export interface WorkSpaceParameter {
  workspacesSearched?: WorkspacesSearched;
  floors?: Floors;
  sections?: Sections;
}

export interface propsSection {
  idFloor: number;
  idBuilding: number;
  name: string;
}

export const useStateContainer = (initialState: WorkSpaceParameter) => {
  const [workplaces, setworkplaces] = useState(
    initialState.workspacesSearched || { workplaces: [] },
  );

  const [floors, setfloors] = useState(
    initialState.floors || { floorsResult: [] },
  );
  const [sections, setsections] = useState(
    initialState.sections || { sectionsResult: [] },
  );

  const [parameters, setparameters] = useState('');
  const [building, setbuilding] = useState('Torre Central 2');
  const [floor, setfloor] = useState('Piso 1');
  const [section, setsection] = useState('Seccion A');
  const [date, setdate] = useState(
    moment(new Date()).add(1, 'days').format('DD/MM/YYYY'),
  );

  const searchWorkplaces = () => {
    ReserveServices.getSearchedWorkplaces().then((resultSearch) => {
      setworkplaces(resultSearch);
    });
  };
  const obtainFloors = () => {
    ReserveServices.getFloors().then((resultFloor) => setfloors(resultFloor));
  };

  const obtainSections = (id?: number) => {
    ReserveServices.getSections().then((resultSections) => {
      setsections(resultSections);
    });
  };

  const changeParameter = (parameter: string, value: string) => {
    if (parameter === 'piso') {
      setfloor(value);
    }
    if (parameter === 'section') {
      setsection(value);
    }
    if (parameter === 'date') {
      setdate(value);
    }
  };

  return {
    data: { workplaces, floors, sections, building, floor, section, date },
    mutations: {
      searchWorkplaces,
      obtainFloors,
      obtainSections,
      changeParameter,
    },
  };
};

export const ReserveContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface SearchWorkspaceProvider {
  initialState?: WorkSpaceParameter;
}

export const ReserveProvider: React.FC<SearchWorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <ReserveContext.Provider value={contextValue}>
      {children}
    </ReserveContext.Provider>
  );
};
