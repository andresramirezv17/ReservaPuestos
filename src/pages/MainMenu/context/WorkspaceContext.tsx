import React, { createContext, useEffect, useState } from 'react';
import { UserWorkspace } from './models/UserWorkspace';
import * as WorkspaceService from '../services/WorkspaceServices';

export interface WorkSpaceState {
  workSpace?: UserWorkspace;
  varpueba?: boolean;
}

export const useStateContainer = (initialState: WorkSpaceState) => {
  const [workspaceUser, setworkspaceUser] = useState(
    initialState.workSpace || {},
  );
  const [prueba, setprueba] = useState(initialState.varpueba || true);

  useEffect(() => {
    WorkspaceService.getReserves(1).then((reserves) =>
      setworkspaceUser(reserves),
    );
  }, []);

  return { workspaceUser, prueba };
};

export const WorkspaceContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface WorkspaceProvider {
  initialState?: WorkSpaceState;
}

export const UserWorkspaceProvider: React.FC<WorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <WorkspaceContext.Provider value={contextValue}>
      {children}
    </WorkspaceContext.Provider>
  );
};
