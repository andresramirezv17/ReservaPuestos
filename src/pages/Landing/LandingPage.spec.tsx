import React from 'react';
import { LandingPage } from './index';
import { render } from '@testing-library/react';
import { SessionProvider } from '../../context/SessionContext';

const defaultProps: any = {
  history: {
    replace: jest.fn(),
  },
};

describe('Landing test', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <SessionProvider>
        <LandingPage {...defaultProps} />
      </SessionProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
