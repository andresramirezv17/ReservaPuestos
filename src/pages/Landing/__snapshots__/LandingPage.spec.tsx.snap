// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing test should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root makeStyles-container-1 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="makeStyles-containerImg-2"
          >
            <img
              alt="Logo"
              class="makeStyles-img-3"
              src="workspace.png"
            />
          </div>
          <p
            class="MuiTypography-root makeStyles-title-4 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            Bienvenido
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            Ingresa para poder ver tus puestos de trabajo
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-input-5 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              User
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="input-username"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    User
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inputPass-6 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="password-username"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="continue"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Continue
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
